spec/lib/api/v3/queries/query_representer_spec.rb

```diff
@@ -0,0 +1,147 @@
+
+  describe 'generation' do
+    describe '_links' do
+      it_behaves_like 'has a titled link' do
+        let(:link) { 'self' }
+        let(:href) { api_v3_paths.query query.id }
+        let(:title) { query.name }
+      end
+
+      it_behaves_like 'has a titled link' do
+        let(:link) { 'user' }
+        let(:href) { api_v3_paths.user query.user_id }
+        let(:title) { query.user.name }
+      end
+
+      it_behaves_like 'has a titled link' do
+        let(:link) { 'project' }
+        let(:href) { api_v3_paths.project query.project_id }
+        let(:title) { query.project.name }
+      end
+
+      context 'has no project' do
+        let(:query) { FactoryGirl.build_stubbed(:query, project: nil) }
+
+        it_behaves_like 'has an empty link' do
+          let(:link) { 'project' }
+        end
+      end
+    end
+
+    it 'should show an id' do
+      is_expected.to be_json_eql(query.id).at_path('id')
+    end
+
+    it 'should show the query name' do
+      is_expected.to be_json_eql(query.name.to_json).at_path('name')
+    end
+
+    it 'should indicate whether sums are shown' do
+      is_expected.to be_json_eql(query.display_sums.to_json).at_path('displaySums')
+    end
+
+    it 'should indicate whether the query is publicly visible' do
+      is_expected.to be_json_eql(query.is_public.to_json).at_path('isPublic')
+    end
+
+    describe 'grouping' do
+      let(:query) { FactoryGirl.build_stubbed(:query, group_by: 'assigned_to') }
+
+      it 'should show the grouping column' do
+        is_expected.to be_json_eql('assignee'.to_json).at_path('groupBy')
+      end
+
+      context 'without grouping' do
+        let(:query) { FactoryGirl.build_stubbed(:query, group_by: nil) }
+
+        it 'should show no grouping column' do
+          is_expected.to be_json_eql(nil.to_json).at_path('groupBy')
+        end
+      end
+    end
+
+    describe 'with filters' do
+      let(:query) {
+        FactoryGirl.build_stubbed(:query,
+                                  filters: [
+                                    Queries::WorkPackages::Filter.new('status_id',
+                                                                      operator: '=',
+                                                                      values: ['1'])
+                                  ])
+      }
+
+      it 'should render the filters' do
+        expected = [
+          {
+            status: {
+              operator: '=',
+              values: ['1']
+            }
+          }
+        ]
+        is_expected.to be_json_eql(expected.to_json).at_path('filters')
+      end
+    end
+
+    describe 'with sort criteria' do
+      let(:query) {
+        FactoryGirl.build_stubbed(:query,
+                                  sort_criteria: [['subject', 'asc'], ['assigned_to', 'desc']])
+      }
+
+      it 'should render the filters' do
+        is_expected.to be_json_eql([
+                                     ['subject', 'asc'],
```

Comment:
Use 2 spaces for indentation in an array, relative to the start of the line where the left bracket is.